packaging/Microsoft.NET.Sdk.WindowsDesktop/targets/Microsoft.NET.Sdk.WindowsDesktop.props

```diff
@@ -5,6 +5,10 @@
                            Condition="'$(EnableDefaultApplicationDefinition)' != 'false' And Exists('$(MSBuildProjectDirectory)/App.xaml')">
       <Generator>MSBuild:Compile</Generator>
     </ApplicationDefinition>
+    <ApplicationDefinition Include="Application.xaml"
```

Comment:
Consider adding an additional project extension condition.  Each ApplicationDefinition would be scoped to a specific language to prevent duplicate ApplicationDefinitions.   

```xml
 <ApplicationDefinition Include="App.xaml"

                           Condition="'$(EnableDefaultApplicationDefinition)' != 'false' And Exists('$(MSBuildProjectDirectory)/App.xaml') And '$(MSBuildProjectExtension)' == '.csproj'">

      <Generator>MSBuild:Compile</Generator>

    </ApplicationDefinition>

    <ApplicationDefinition Include="Application.xaml"

                           Condition="'$(EnableDefaultApplicationDefinition)' != 'false' And Exists('$(MSBuildProjectDirectory)/Application.xaml') And '$(MSBuildProjectExtension)' == '.vbproj'">

      <Generator>MSBuild:Compile</Generator>

    </ApplicationDefinition>
```